lib/SILOptimizer/Mandatory/TFDeabstraction.cpp

```diff
@@ -237,16 +225,6 @@ void TFDeabstraction::inlineCalls() {
     if (callee.getInlineStrategy() == NoInline)
       return false;
 
-    // In dynamic compliation mode, do not inline functions from the same
-    // module. (We still inline functions from other modules, because there is a
-    // risk that the other module was not compiled in dynamic compilation mode
-    // and therefore its functions are not deabstracted or lowered. For
-    // example, we do not compile the standard library in dynamic compilation
-    // mode.)
-    if (llvm::TFDynamicCompilation && !isAcceleratorOnly(fn) &&
```

Comment:
I added a summary of the simplifications and why they're correct. This is the `shouldBeInlined` one.

I think that I have only removed "dead" conditions that no longer fire, because none of the conditions that I removed fire in non-dynamic mode.

Reply:
thanks!

> because none of the conditions that I removed fire in non-dynamic mode.

In non-dynamic mode, do they not fire because we don't have unit test coverage, or will not never fire based on (static) code analysis?

It's not obvious to me that we are in the latter case, though I don't yet have a concrete test case in graph mode to confirm on this point.

Reply:
Never, based on static code analysis. All the conditions that I removed are some variant of `if (llvm::TFDynamicCompilation && ...)` so they don't fire when `llvm::TFDynamicCompilation` is false. Except the `fn.isAvailableExternally()` one, but that one gets handled in the `tfc.shouldBePartitioned` (described more in PR desc).

Reply:
Great, thanks!